docker/memcached.conf

```diff
@@ -0,0 +1,46 @@
+# memcached config file
+#
+# You can insert into this config file all options of memcached. Please 
+# refer the manpage.
+# It will be appended to the memcached command on startup of the container
+
+# Be verbose
+# -v
+
+# Be even more verbose (print client commands as well)
+# -vv
+
+# Start with a cap of 64 megs of memory. It's reasonable, and the daemon default
+# Note that the daemon will grow to this size, but does not start out holding this much
+# memory
+-m 64
+
+# Default connection port is 11211
+-p 11211
+
+# Run the daemon as root. The start-memcached will default to running as root if no
+# -u command is present in this config file
+-u memcache
+
+# Specify which IP address to listen on. The default is to listen on all IP addresses
+# This parameter is one of the only security measures that memcached has, so make sure
+# it's listening on a firewalled interface.
+-l 127.0.0.1
+
+# Limit the number of simultaneous incoming connections. The daemon default is 1024
+# -c 1024
+
+# Lock down all paged memory. Consult with the README and homepage before you do this
+# -k
+
+# Return error when memory is exhausted (rather than removing items)
+# -M
+
+# Maximize core file limit
+# -r
+
+# Override the default size of each slab page. Default is 1mb. Default is 1m, minimum
+# is 1k, max is 128m. Adjusting this value changes the item size limit. Beware that
+# this also increases the number of slabs (use -v to view), and the overal memory usage
+# of memcached.
+-I 1m 
```

Comment:
Might be worth making this an environment variable with a default so it can be overridden without having to make a new image.

Reply:
With the config file solution it is now possible to configure all options of memcache with the file like in every debian environment it is possible. 

When we switch to environment variables it is a little bit more complicated in my opinion. Should we use one environment variable per switch? Then it is no longer possible to use every switch that memcache has, only that one for that we provide environment variables.
Other option is one environment variable for all switches. This is could be confusing in configuration because some characters has to be parsed. 

So both is not the best solution. 

In my opinion the conf file is the best one. We are in the debian standard, and you don't have to create every time a new image when you want to change the config. 
You can easily use the docker volume feature to mount your own memcached.conf file into the container. See [https://stackoverflow.com/questions/42248198/how-to-mount-a-single-file-in-a-volume](this stackoverflow post for a howto)
This is how many other projects also do the configuration management and a proven way in practice. 
For example we need also a different -I value for our environment and I do it the same way. 

just my two cents...